modular_skyrat/modules/modular_implants/code/soulcatcher/soulcatcher_mob.dm

```diff
@@ -0,0 +1,173 @@
+	outside_sight = !outside_sight
+	if(outside_sight)
+		cure_blind(NO_EYES)
+	else
+		become_blind(NO_EYES)
+
+	return outside_sight
+
+/// Toggles whether or not the soul inside the soulcatcher can see the outside world. returns the state of the `outside_sight` variable.
+/mob/living/soulcatcher_soul/proc/toggle_hearing()
+	outside_hearing = !outside_hearing
+	if(outside_hearing)
+		REMOVE_TRAIT(src, TRAIT_DEAF, INNATE_TRAIT)
+	else
+		ADD_TRAIT(src, TRAIT_DEAF, INNATE_TRAIT)
+
+	return outside_sight
+
+/// Attemp to leave the soulcatcher.
+/mob/living/soulcatcher_soul/verb/leave_soulcatcher()
+	set name = "Leave Soulcatcher"
+	set category = "IC"
+
+	if(!able_to_leave)
+		to_chat(src, span_warning("You are unable to leave the soulcatcher."))
+		return FALSE
+
+	if(tgui_alert(src, "Are you sure you wish to leave the soulcatcher?", "Soulcatcher", list("Yes", "No")) != "Yes")
+		return FALSE
+
+	qdel(src)
+
+/mob/living/soulcatcher_soul/ghost()
+	. = ..()
+	qdel(src)
+
+/mob/living/soulcatcher_soul/say(message, bubble_type, list/spans, sanitize, datum/language/language, ignore_spam, forced, filterproof, message_range, datum/saymode/saymode)
+	message = trim(copytext_char(sanitize(message), 1, MAX_MESSAGE_LEN))
+	if(!message || message == "")
+		return
+
+	if(!able_to_speak)
+		to_chat(src, span_warning("You are unable to speak!"))
+		return FALSE
+
+	var/datum/soulcatcher_room/room = current_room.resolve()
+	if(!room)
+		return FALSE
+
+	room.send_message(message, src, FALSE)
+	return TRUE
+
+/mob/living/soulcatcher_soul/me_verb(message as text)
+	message = trim(copytext_char(sanitize(message), 1, MAX_MESSAGE_LEN))
+	if(!message)
+		return FALSE
+
+	if(!able_to_emote)
+		to_chat(src, span_warning("You are unable to speak!"))
+		return FALSE
+
+	var/datum/soulcatcher_room/room = current_room.resolve()
+	if(!room)
+		return FALSE
+
+	room.send_message(message, src, TRUE)
+	return TRUE
+
+/mob/living/soulcatcher_soul/subtle()
+	set hidden = TRUE
+	return FALSE
+
+/mob/living/soulcatcher_soul/subtler()
+	set hidden = TRUE
+	return FALSE
+
+/mob/living/soulcatcher_soul/whisper_verb()
+	set hidden = TRUE
+	return FALSE
+
+/mob/living/soulcatcher_soul/container_emote()
+	set hidden = TRUE
+	return FALSE
+
+/mob/living/soulcatcher_soul/resist()
+	set hidden = TRUE
+	return FALSE
+
+/mob/living/soulcatcher_soul/Destroy()
+	if(current_room)
+		var/datum/soulcatcher_room/room = current_room.resolve()
+		log_admin("[src] left the soulcatcher room, [room].")
+		room.current_souls -= src
+
+	if(previous_body && mind)
+		var/mob/target_body = previous_body.resolve()
+		mind.transfer_to(target_body)
+		var/datum/component/previous_body/body_component = target_body.GetComponent(/datum/component/previous_body) //Is the soul currently within a soulcatcher?
+		if(body_component)
+			body_component.restore_mind = FALSE
+			qdel(body_component)
+
+		if(target_body.stat != DEAD)
+			target_body.grab_ghost(TRUE)
+
+	return ..()
+
+/datum/component/previous_body
```

Comment:
Also the component itself should have at least a little bit of documentation, telling us what it does.